### DISCUSSION TITLE
error using colmap2nerf.py

### DISCUSSION THREAD

**Discussion by Hurleyworks** [NVlabs/instant-ngp#165] [Q&A]
Howdy,  thanks for making this amazing project available!

I'm on Windows 10 with a GeForce RTX 2070 SUPER. I'm having no problems at all compiling and running all the instant-npg testbed examples. But I am unable to run the colmap2nerf.py script successfully. 

**I have:**
Installed Python 3.10
Installed  numpy and opencv-python modules.
downloaded the latest Pre-Release version of colmap and added it's bin folder to my Path ( I also tried the stable release version with same result)

I have placed the images folder containing the Fox example images in the same directory where colmap2nerf.py lives.

Here's the output when I run the script. Any ideas what I might be doing wrong?. Unfortunately I have little experience with Python.

PS D:\ActiveWorks\test> python colmap2nerf.py --colmap_matcher exhaustive --run_colmap --aabb_scale 16
running colmap with:
        db=colmap.db
        images=images
        sparse=colmap_sparse
        text=colmap_text
warning! folders 'colmap_sparse' and 'colmap_text' will be deleted/replaced. continue? (Y/n)y
==== running: colmap feature_extractor --ImageReader.camera_model OPENCV --ImageReader.single_camera 1 --database_path colmap.db --image_path images
FATAL: command failed

**Comment by Cortex78** [comment]
Hi,

i'm running into the same error.

Did you found a way ?

Thanks

**Comment by kajc10** [comment]
is colmap on the PATH correctly? to check it try running `colmap --help`

**Comment by hanikhatib** [comment]
This might be a matter of how you configured your directory. In the Colmap documentation, it asks that the images are placed in a subfolder called 'images', with the folder being the directory for other files that are written in colmap, such as the transforms.json, colmap.db and others. First, ensure your libraries are on the right path and second, make sure you run the command from the images folder directory (e.g. cd _directory_name_) with the subfolder called 'images' containing the images. Very late response but hope that helps!

**Reply by HannahHaensen** [reply]
hey can you maybe post an screenshot of your folder structure?

![image](https://user-images.githubusercontent.com/22636930/186693626-28067e3b-b148-4096-ac8e-74128403520d.png)

i have it like this and run into the same issue when running:
```
==== running: colmap feature_extractor --ImageReader.camera_model OPENCV --ImageReader.camera_params "" --SiftExtraction.estimate_affine_shape=true --SiftExtraction.domain_size_pooling=true
 --ImageReader.single_camera 1 --database_path colmap.db --image_path "images"
FATAL: command failed
```
when running the command directly in the shell the error does not appear.


```
(venv) C:\Users\user\Documents\GitHub\instant-ngp\data\test>python ../../scripts/colmap2nerf.py --colmap_matcher exhaustive --run_colmap --aabb_scale 16
running colmap with:
        db=colmap.db
        images="images"
        sparse=colmap_sparse
        text=colmap_text
warning! folders 'colmap_sparse' and 'colmap_text' will be deleted/replaced. continue? (Y/n)y
FEATURE EXTRACTOR
==== running: colmap feature_extractor --ImageReader.camera_model OPENCV --ImageReader.camera_params "" --SiftExtraction.estimate_affine_shape=true --SiftExtraction.domain_size_pooling=true
 --ImageReader.single_camera 1 --database_path colmap.db --image_path "images"
FATAL: command failed
```

**Comment by timi-liuliang** [comment]
1.Go into colmap directory.
2.copy all files in lib directory to bin directory
Then the problem will resolved.

I can't understand why colmap don't put *.dll and *.exe files together.
